real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/planners/WayPointFollowerPlanner.java

```diff
@@ -32,75 +49,219 @@ public void updatePositionEstimate(GPSPoseMessage m) {
             return 0;
         }
 
+        //find a path from our current location to that point
+        GpsMeasurement target = wayPoints.get(lookaheadIndex);
+        currentWaypoint = target;
+        double dx = LocalizerUtil.convertLonToMeters(target.getLongitude()) - LocalizerUtil.convertLonToMeters(pose.getLongitude());
+        double dy = LocalizerUtil.convertLatToMeters(target.getLatitude()) - LocalizerUtil.convertLatToMeters(pose.getLatitude());
+        double deltaHeading = Math.atan2(dy, dx) - pose.getHeading();
 
-        GpsMeasurement targetPoint = wayPoints.get(targetIndex);
+        GpsMeasurement B = wayPoints.get(closestIndex);
+        GpsMeasurement A = wayPoints.get(closestIndex + 1);
+        GPSPoseMessage P = pose;
 
-        //find a path from our current location to that point
-        double dLon = targetPoint.getLongitude() - pose.getLongitude();
-        double dLat = targetPoint.getLatitude() - pose.getLatitude();
-        double desiredHeading = Math.toDegrees(Math.atan2(LocalizerUtil.convertLatToMeters(dLat), LocalizerUtil.convertLonToMeters(dLon)));
+        double padx = LocalizerUtil.convertLonToMeters(A.getLongitude()) - LocalizerUtil.convertLonToMeters(P.getLongitude());
+        double pady = LocalizerUtil.convertLatToMeters(A.getLatitude()) - LocalizerUtil.convertLatToMeters(P.getLatitude());
+        double phi = Math.atan2(pady, padx);
+
+        double badx = LocalizerUtil.convertLonToMeters(A.getLongitude()) - LocalizerUtil.convertLonToMeters(B.getLongitude());
+        double bady = LocalizerUtil.convertLatToMeters(A.getLatitude()) - LocalizerUtil.convertLatToMeters(B.getLatitude());
+        double psi = Math.atan2(bady, badx);
+
+        double theta = phi - psi;
 
-        // basically we want all of our angles to be in the same range, so that we don't
-        // have weird wraparound
-        desiredHeading = Util.normalizeAngleDeg(desiredHeading);
-        double poseHeading = Util.normalizeAngleDeg(pose.getHeading());
+        double L = GPSPoseMessage.getDistance(P, B.toGpsPoseMessage(0));
+        double E = L * Math.sin(theta);
 
-        //find the angle we need to reach that point
-        return Util.normalizeAngleDeg(desiredHeading - poseHeading);
+        double thetaDelta = phi - psi;
 
+        if (Math.signum(thetaDelta) != Math.signum(thetaDelta*E/10.0)){
+            E=1;
+        }
+
+        thetaDelta *= E/10.0;
+        
+//        deltaHeading += thetaDelta;
+
+        //Pure Pursuit steering controller
+        double commandedAngle = Math.atan2(2 * RobobuggyKFLocalizer.WHEELBASE_IN_METERS * Math.sin(deltaHeading), lookahead);
+        commandedAngle = Util.normalizeAngleRad(commandedAngle);
+        return commandedAngle;
     }
 
-    @Override
-    protected boolean getDeployBrakeValue() {
-        return false;
-        /*
-		  int closestIndex = getClosestIndex(wayPoints,pose);
+    private double stanleyMethodController() {
```

Comment:
I went ahead and removed this since I was cleaning up the rest of this file anyway